src/sdsc/__init__.py

```diff
@@ -1009,6 +1009,42 @@ def splitpath(context, path, wantedsegment='filename'):
     else:
         return path
 
+def splitvalueunit(context, value, part='value'):
+    """ From a combination of numeric value and unit (like "100px"), returns
+    either the value or the unit.
+
+    :param str context: context node (ignored)
+    :param str value: numeric value and unit
+    :param str part: wanted part of input value: 'value' or 'unit'
+    :return: numeric value or unit of input value
+    """
+
+    del context # not used
+
+    value = str(value)
+    part = str(part)
+
+    regex_valueformat = re_compile(r'[.\d]+([a-z]{2,3}|%)?')
+    regex_numericvalue = re_compile(r'^[.\d]+')
+    regex_unit = re_compile(r'([a-z]{1,3}|%)$')
+
+    if not value:
+        return "MALFORMED_VALUE"
+    if not regex_valueformat.fullmatch(value):
+        return "MALFORMED_VALUE"
```

Comment:
Hm, "nan" is indeed a valid value for float in Python (however, only in lowercase -- if you enter it in uppercase, it will be converted).

Anyway, I don't think I want to change this to an "NaN" string, necessarily -- neither my (valid) input nor my output is necessarily a number: Though both can be a number, the input usually has some kind of unit and the output can be a unit.